types/iopipe__iopipe/index.d.ts

```diff
@@ -0,0 +1,24 @@
+// Type definitions for iopipe__iopipe 1.12
+// Project: https://github.com/iopipe/iopipe (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
+// Definitions by: Javon Harper <https://github.com/javonharper>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export interface LibraryConfig {
+    debug?: boolean;
+    token?: string;
+    networkTimeout?: number;
+    timeoutWindow?: number;
+}
+
+export type FunctionWrapper = (handler: any) => void;
+
+export function label(label: string): void;
+
+export function metric(label: string, value: number): void;
+
+export namespace mark {
+    function start(label: string): void;
+    function end(label: string): void;
+}
+
+export default function iopipe(config?: LibraryConfig): FunctionWrapper;
```

Comment:
This package was declared using `module.exports = ...` or some variant. It should be declared with an `export = ...` like one of the following:

### Variables

```ts
declare var xyz: SomeType;

export = xyz;
```

### Classes

```ts
class Foo {
  // ...
}

export = Foo
```

### Functions

```ts
function foo(): any;

export = foo;
```

To import the entity using a default import, you will need to enable `esModuleInterop`. Alternatively, you can just import using a more appropriate syntax like

```ts
import foo = require("foo");
```

[See more here in the FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9f6430117d025aef62ed56128deb65508edc86/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default).

Reply:
When I change the syntax I get this error: `An export assignment cannot be used in a module with other exported elements. [2309]`

The code looks like:
```
declare function iopipe(config?: LibraryConfig): FunctionWrapper;

export = iopipe

```

am I doing something wrong? Maybe I need to change how I use my named exports?

Reply:
FYI I'd like the export to be able to look something like `import iopipe, { label } from "@iopipe/iopipe"` but I think the default export is clashing with the named exports atm. Thanks!

Reply:
I was able to get it compiling by declaring a namespace alongside the member function and enabling `allowSyntheticDefaultImports`.

Reply:
I was able to get it compiling by declaring a namespace alongside the member function and enabling `allowSyntheticDefaultImports`.